src/ProvidedTypes.fs

```diff
@@ -14097,6 +14097,8 @@ namespace ProviderImplementation.ProvidedTypes
 
                     match ptdT.GetConstructors(bindAll) |> Seq.tryPick (function :? ProvidedConstructor as pc when pc.IsTypeInitializer -> Some pc | _ -> None) with
                     | None -> ()
+                    | Some _ when ptdT.IsInterface ->
+                        failwith "The provided type definition is an interface; therefore, it may not provide constructors."

```

Comment:
I added this one check for provided constructors for interfaces. Abstract classes can still have constructors, so I left it to interfaces alone.